File: src/main/java/ru/stnk/RestTestAPI/repository/QuoteBinOneMinuteRepository.java
package ru.stnk.RestTestAPI.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.stnk.RestTestAPI.entity.coins.QuoteBinOneMinute;

import java.util.List;

@Repository
public interface QuoteBinOneMinuteRepository extends JpaRepository<QuoteBinOneMinute, Long> {
    QuoteBinOneMinute findBySymbolName(String symbolName);
    List<QuoteBinOneMinute> findBySymbolNameOrderByTimestampDesc(String symbolName);
}

File: src/main/kotlin/ru/stnk/resttestapi/repository/QuoteBinOneMinuteRepository.kt
package ru.stnk.resttestapi.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.stnk.resttestapi.entity.coins.QuoteBinOneMinute

@Repository
interface QuoteBinOneMinuteRepository : JpaRepository<QuoteBinOneMinute, Long> {
    fun findBySymbolName(symbolName: String): QuoteBinOneMinute?
    fun findBySymbolNameOrderByTimestampDesc(symbolName: String): List<QuoteBinOneMinute>?
}
